### PR TITLE
SLING-10988 provide web console plugin for initial content

### PR DESCRIPTION
**Pull request by kwin**

### CODE REVIEW

File: src/main/java/org/apache/sling/jcr/contentloader/internal/ContentLoaderWebConsolePlugin.java [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+@Component(service=javax.servlet.Servlet.class,
+    property = {
+        Constants.SERVICE_VENDOR + "=The Apache Software Foundation",
+        Constants.SERVICE_DESCRIPTION + "=Apache Sling JCR Content Loader Web Console Plugin",
+        "felix.webconsole.label=" + ContentLoaderWebConsolePlugin.LABEL,
+        "felix.webconsole.title=JCR Content Loader",
+        "felix.webconsole.category=Sling"
+    })
+@SuppressWarnings("serial")
+public class ContentLoaderWebConsolePlugin extends GenericServlet {
```
Comment (stefanseifert) [inline src/main/java/org/apache/sling/jcr/contentloader/internal/ContentLoaderWebConsolePlugin.java:58]: i've deployed this in a local AEMaaCS SDK instance and was not able to see the console.
do we have to extend AbstractWebConsolePlugin?

Comment (kwin) [reply src/main/java/org/apache/sling/jcr/contentloader/internal/ContentLoaderWebConsolePlugin.java:58]: I just tried this as well and there the component "org.apache.sling.jcr.contentloader.internal.BundleContentLoaderListener" is unsatisfied.
Screenshot 2021-12-10 at 17 32 47

I think this is rather related to https://github.com/apache/sling-org-apache-sling-jcr-contentloader/commit/235a16eaa8cb6ed0fde56b1966e6506d1edae46f

Comment (kwin) [reply src/main/java/org/apache/sling/jcr/contentloader/internal/ContentLoaderWebConsolePlugin.java:58]: AEMaaCS ships with the old version 2.3.0 which still relies on admin sessions (https://issues.apache.org/jira/browse/SLING-9735) and hence does not have any service user configured for jcr.contentloader.

**Comment by kwin** [comment]
@oliverlietz All paxexam tests are failing for some reason and I cannot see an obvious reason in the 3MB log in `target/failsafe-reports/org.apache.sling.jcr.contentloader.it.BasicInitialContentIT-output.txt`. I don't have much experience with paxexam unfortunately and there is not even log levels emitted by default...

**Comment by oliverlietz** [comment]
@kwin Was the test content moved or bundle symbolic name changed? On console I can see:

```
[ERROR] org.apache.sling.jcr.contentloader.it.BasicInitialContentIT.folderWithDescriptor  Time elapsed: 10.009 s  <<< ERROR!
org.ops4j.pax.swissbox.tracker.ServiceLookupException: gave up waiting for service org.apache.sling.resource.presence.ResourcePresence
```

Path is defined in ``ContentloaderTestSupport`` : ``protected static final String CONTENT_ROOT_PATH = "/test-content/" + BUNDLE_SYMBOLICNAME;``

**Comment by kwin** [comment]
It must be a change in this PR as master works fine. No test content has been moved

**Comment by rombert** [comment]
I was trying to understand whether the failure is caused by the new bundle requirements ( servlet-api, sling.api ), but I can't find any clues. Maybe someone with a better understanding of the test framework can tell whether the bundle is active or not. AFAICT, since we wait for the `ResourcePresence` reference, we fail fast before trying to assert that the bundle is active.

**Comment by kwin** [comment]
@rombert As said in https://issues.apache.org/jira/browse/SLING-10988?focusedCommentId=17458301&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-17458301 this is just an issue with registering BundleContentLoaderListener as BundleHelper. It works without that service registration for some reason...

**Comment by kwin** [comment]
> It works without that service registration for some reason...

Found the reason: The component was implicitly immediate without any service registration. With at least one service registration the default is lazy, i.e. that component became no longer active. This is fixed now in https://github.com/apache/sling-org-apache-sling-jcr-contentloader/pull/8/commits/18b25d5f431e61eb999bfd69029466b67a99bac0 which also fixed the PaxExam ITs.

**Review by rombert — COMMENTED** [review]
LGTM, thanks.

**Pull request merged**